lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -77,9 +77,20 @@ def _get(self, json_data: JSON, request: Request) -> Response:
         name = json.get("name")
         keys = json.get("metadata")
 
-        self.logger.info("GET metadata {} for {}", keys, name)
         try:
-            metadata = self._get_metadata(name, keys)
+            dataset = Dataset.query(name=name)
+        except DatasetError as e:
+            self.logger.warning("Dataset {!r} not found: {}", name, str(e))
+            raise APIAbort(HTTPStatus.BAD_REQUEST, f"Dataset {name!r} not found")
```

Comment:
Since this a 400-level error (i.e., a "user error"), are you sure we want to make a log entry for it?  If so, is there any other information that we want to log for context (like some sort of requestor ID information)?  (I note that we don't log anything anywhere else in this function.)

Ditto line 136.

Reply:
I think I cut and pasted that from somewhere else; we have logged a lot of client errors as "warning". It's a fair question to what extent we want to do that; it's mostly useful during debugging.

Reply:
> it's mostly useful during debugging.

Right...but, in that case, they should probably be logged as debug-level messages -- that way we can shut them off or easily delete them from the code.  (I would rather avoid having to do a wholesale pass through the code later, looking for log messages which should be removed for production.)